File: leon-samples/ajax/reverser/java_js/src/main/java/io/leon/samples/ajax/reverser/java_js/ReverserService.java
package io.leon.samples.ajax.reverser.java_js;


public class ReverserService {

    public String reverse(String text, boolean toUpperCase) {


        /*
        try {
            Thread.sleep(5000);
        } catch (InterruptedException e) {
            e.printStackTrace();  //To change body of catch statement use File | Settings | File Templates.
        }
        */


        String reversed = new StringBuffer(text).reverse().toString();
        if (toUpperCase)
            return reversed.toUpperCase();
        else
            return reversed;
    }
}

File: leon-samples/ajax/reverser/js_js/src/main/resources/ReverserService.js

var ReverserService = (function() {

    var logger = leon.getLogger("ReverserService");

    var StringBuffer = Packages.java.lang.StringBuffer;

    return {
        reverse: function(text, toUpperCase) {
            var sb = new StringBuffer(text);
            var reversed = sb.reverse().toString();
            if (toUpperCase) {
                return reversed.toUpperCase();
            } else {
                return reversed;
            }
        }
    };
})();

